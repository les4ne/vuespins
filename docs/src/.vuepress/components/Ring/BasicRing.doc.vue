<template>
  <div>
    <Ring />
  </div>
</template>

<script>
import { Ring } from 'vuespins'

export default {
  name: 'app',
  components: { Ring },
}
</script>
