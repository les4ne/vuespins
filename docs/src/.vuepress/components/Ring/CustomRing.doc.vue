<template>
  <div>
    <Ring color="#fb503b" :size="80" :thickness="1" :speed="3" />
  </div>
</template>

<script>
import { Ring } from 'vuespins'

export default {
  name: 'app',
  components: { Ring }
}
</script>
