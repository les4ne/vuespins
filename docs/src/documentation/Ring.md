# Ring

## **Basic**

<br><Ring />

### Props

Basic "Ring" component

|  property   | propType | required | default | custom |
| :---------: | :------: | :------: | :-----: | :----: |
|   `color`   |  string  |    -     | #282c34 |   -    |
|   `size`    |  string  |    -     |   50    |   -    |
| `thickness` |  string  |    -     |    6    |   -    |

### Usage

<<< @/src/.vuepress/components/Ring/BasicRing.doc.vue

## **Custom**

<br><Ring color="#fb503b" size="80" thickness="1" />

### Props

Custom "Ring" component

|  property   | propType | required | default | custom  |
| :---------: | :------: | :------: | :-----: | :-----: |
|   `color`   |  string  |    -     | #282c34 | #fb503b |
|   `size`    |  string  |    -     |   50    |   80    |
| `thickness` |  string  |    -     |    6    |    1    |

### Usage

<<< @/src/.vuepress/components/Ring/CustomRing.doc.vue
