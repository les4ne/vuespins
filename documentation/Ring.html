<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Ring | vuespins (1.0.4)</title>
    <meta name="generator" content="VuePress 1.5.4">
    
    <meta name="description" content="Vuespins Docs">
    <meta name="theme-color" content="#3eaf7c">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="preload" href="/vuespins/assets/css/0.styles.954e083e.css" as="style"><link rel="preload" href="/vuespins/assets/js/app.1de3f0c3.js" as="script"><link rel="preload" href="/vuespins/assets/js/2.ba357b4e.js" as="script"><link rel="preload" href="/vuespins/assets/js/25.593d58d4.js" as="script"><link rel="prefetch" href="/vuespins/assets/js/10.8f93b35e.js"><link rel="prefetch" href="/vuespins/assets/js/11.bd979311.js"><link rel="prefetch" href="/vuespins/assets/js/12.4d342e5e.js"><link rel="prefetch" href="/vuespins/assets/js/13.0135f541.js"><link rel="prefetch" href="/vuespins/assets/js/14.5d67a5e5.js"><link rel="prefetch" href="/vuespins/assets/js/15.11bb3d13.js"><link rel="prefetch" href="/vuespins/assets/js/16.8689023d.js"><link rel="prefetch" href="/vuespins/assets/js/17.1b4c6820.js"><link rel="prefetch" href="/vuespins/assets/js/18.fc50494c.js"><link rel="prefetch" href="/vuespins/assets/js/19.5f94a2bf.js"><link rel="prefetch" href="/vuespins/assets/js/20.94d4ca68.js"><link rel="prefetch" href="/vuespins/assets/js/21.b1691a83.js"><link rel="prefetch" href="/vuespins/assets/js/22.b23320cf.js"><link rel="prefetch" href="/vuespins/assets/js/23.d2c7bc3e.js"><link rel="prefetch" href="/vuespins/assets/js/24.3a2867c3.js"><link rel="prefetch" href="/vuespins/assets/js/26.b1d739a7.js"><link rel="prefetch" href="/vuespins/assets/js/27.29a9fb28.js"><link rel="prefetch" href="/vuespins/assets/js/28.fc8ffcb7.js"><link rel="prefetch" href="/vuespins/assets/js/3.6baab57e.js"><link rel="prefetch" href="/vuespins/assets/js/4.6f99b75f.js"><link rel="prefetch" href="/vuespins/assets/js/5.312661c6.js"><link rel="prefetch" href="/vuespins/assets/js/6.d6ad6eb7.js"><link rel="prefetch" href="/vuespins/assets/js/7.88b4d6f2.js"><link rel="prefetch" href="/vuespins/assets/js/8.25ad4ed6.js"><link rel="prefetch" href="/vuespins/assets/js/9.e6cbcfdc.js">
    <link rel="stylesheet" href="/vuespins/assets/css/0.styles.954e083e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/vuespins/" class="home-link router-link-active"><!----> <span class="site-name">vuespins (1.0.4)</span></a> <div class="links"><!----> <nav class="nav-links can-hide"><div class="nav-item"><a href="/vuespins/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://github.com/pwndex/vuespins" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/vuespins/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://github.com/pwndex/vuespins" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Documentation</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/vuespins/documentation/Installation.html" class="sidebar-link">Installation</a></li><li><a href="/vuespins/documentation/Ring.html" aria-current="page" class="active sidebar-link">Ring</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vuespins/documentation/Ring.html#basic" class="sidebar-link">Basic</a></li><li class="sidebar-sub-header"><a href="/vuespins/documentation/Ring.html#custom" class="sidebar-link">Custom</a></li></ul></li><li><a href="/vuespins/documentation/CubeGrid.html" class="sidebar-link">CubeGrid</a></li><li><a href="/vuespins/documentation/Spinner.html" class="sidebar-link">Spinner</a></li><li><a href="/vuespins/documentation/Chase.html" class="sidebar-link">Chase</a></li><li><a href="/vuespins/documentation/Pulse.html" class="sidebar-link">Pulse</a></li><li><a href="/vuespins/documentation/Square.html" class="sidebar-link">Square</a></li><li><a href="/vuespins/documentation/RectGrid.html" class="sidebar-link">RectGrid</a></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="ring"><a href="#ring" class="header-anchor">#</a> Ring</h1> <h2 id="basic"><a href="#basic" class="header-anchor">#</a> <strong>Basic</strong></h2> <br> <!----> <h3 id="props"><a href="#props" class="header-anchor">#</a> Props</h3> <p>Basic &quot;Ring&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>thickness</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">6</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>speed</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">1.2</td> <td style="text-align:center;">-</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Ring</span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> Ring <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> Ring <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h2 id="custom"><a href="#custom" class="header-anchor">#</a> <strong>Custom</strong></h2> <br> <!----> <h3 id="props-2"><a href="#props-2" class="header-anchor">#</a> Props</h3> <p>Custom &quot;Ring&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">#fb503b</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">80</td></tr> <tr><td style="text-align:center;"><code>thickness</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">6</td> <td style="text-align:center;">1</td></tr> <tr><td style="text-align:center;"><code>speed</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">1.2</td> <td style="text-align:center;">3</td></tr></tbody></table> <h3 id="usage-2"><a href="#usage-2" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Ring</span> <span class="token attr-name">color</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>#fb503b<span class="token punctuation">&quot;</span></span> <span class="token attr-name">:size</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>80<span class="token punctuation">&quot;</span></span> <span class="token attr-name">:thickness</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>1<span class="token punctuation">&quot;</span></span> <span class="token attr-name">:speed</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>3<span class="token punctuation">&quot;</span></span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> Ring <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> Ring <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div> <footer class="page-edit"><div class="edit-link"><a href="https://github.com/pwndex/vuespins/edit/master/docs/src/documentation/Ring.md" target="_blank" rel="noopener noreferrer">Edit this page on GitHub</a> <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></div> <div class="last-updated"><span class="prefix">Last Updated:</span> <span class="time">9/26/2020, 12:30:01 AM</span></div></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/vuespins/documentation/Installation.html" class="prev">
        Installation
      </a></span> <span class="next"><a href="/vuespins/documentation/CubeGrid.html">
        CubeGrid
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/vuespins/assets/js/app.1de3f0c3.js" defer></script><script src="/vuespins/assets/js/2.ba357b4e.js" defer></script><script src="/vuespins/assets/js/25.593d58d4.js" defer></script>
  </body>
</html>
