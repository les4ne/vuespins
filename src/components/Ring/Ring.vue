<template>
  <div class="lds-ring" :style="cssVars">
    <div></div>
    <div></div>
    <div></div>
    <div></div>
  </div>
</template>

<script>
export default {
  name: 'Ring',
  props: {
    color: {
      type: String,
      default: '#333',
    },
    size: {
      type: String,
      default: '50',
    },
    thickness: {
      type: String,
      default: '6',
    },
  },
  computed: {
    cssVars() {
      return {
        '--vcs-ring-color': this.color,
        '--vcs-ring-size': this.size + 'px',
        '--vcs-ring-thickness': this.thickness + 'px',
      }
    },
  },
}
</script>

<style scoped>
.lds-ring {
  position: relative;
  width: var(--vcs-ring-size);
  height: var(--vcs-ring-size);
}
.lds-ring div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  border: var(--vcs-ring-thickness) solid var(--vcs-ring-color);
  border-radius: 50%;
  animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: var(--vcs-ring-color) transparent transparent transparent;
}
.lds-ring div:nth-child(1) {
  animation-delay: -0.45s;
}
.lds-ring div:nth-child(2) {
  animation-delay: -0.3s;
}
.lds-ring div:nth-child(3) {
  animation-delay: -0.15s;
}
@keyframes lds-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>