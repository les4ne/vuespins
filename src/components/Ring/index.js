import Ring from './Ring.vue'

export default Ring
