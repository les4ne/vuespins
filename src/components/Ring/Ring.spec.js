import { shallowMount } from '@vue/test-utils'
import Ring from './Ring.vue'

describe('Ring.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(Ring).element).toMatchSnapshot()
  })
})
