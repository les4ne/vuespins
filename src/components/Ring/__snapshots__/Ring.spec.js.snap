// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ring.vue renders correctly 1`] = `
<div
  class="lds-ring"
  style="--vcs-ring-color: #282c34; --vcs-ring-size: 50px; --vcs-ring-thickness: 6px;"
>
  <div />
   
  <div />
   
  <div />
   
  <div />
</div>
`;
